agent/command/results_native.go

```diff
@@ -121,3 +125,35 @@ func (c *attachResults) Execute(ctx context.Context,
 
 	return sendTestResults(ctx, comm, logger, conf, nativeResults.convertToService())
 }
+
+func (c *attachResults) sendTestLogs(ctx context.Context, conf *internal.TaskConfig, logger client.LoggerProducer, comm client.Communicator, results *nativeTestResults) error {
+	logger.Execution().Info("Posting test logs...")
+	for i, res := range results.Results {
+		if err := ctx.Err(); err != nil {
+			return errors.Wrap(err, "operation canceled")
+		}
+
+		if res.LogRaw != "" {
+			testLog := &testlog.TestLog{
+				// When sending test logs we need to use a
+				// unique string since there may be duplicate
+				// log paths if there are duplicate test names.
+				Name:          utility.RandomString(),
+				Task:          conf.Task.Id,
+				TaskExecution: conf.Task.Execution,
+				Lines:         strings.Split(res.LogRaw, "\n"),
+			}
+
+			if err := taskoutput.AppendTestLog(ctx, comm, &conf.Task, testLog); err != nil {
+				// Continue on error to let other logs be
+				// posted.
+				logger.Execution().Error(errors.Wrap(err, "sending test logs"))
+			} else {
+				results.Results[i].LogInfo = &testresult.TestLogInfo{LogName: testLog.Name}
```

Comment:
Can we use res instead of results.Results[i]?

Reply:
This is one of those go traps, range returns a copy of the value not a reference. 

https://golang50shad.es/index.html#range_val_update

Reply:
aaaah thank you!